src/QoiSharp/QoiEncoder.cs

```diff
@@ -15,6 +15,12 @@ public static class QoiEncoder
     /// <returns>Encoded image.</returns>
     /// <exception cref="QoiEncodingException">Thrown when image information is invalid.</exception>
     public static byte[] Encode(QoiImage image)
+    {
+        var bytes = new byte[QoiCodec.HeaderSize + QoiCodec.Padding.Length + (image.Width * image.Height * (byte)image.Channels)];
```

Comment:
It would be nice to expose this calculation as public API so end-users can easily calculate the minimum buffer size needed to be guaranteed to be able to contain the image.